pkg/tfbridge/provider.go

```diff
@@ -1329,7 +1329,7 @@ func (p *Provider) Delete(ctx context.Context, req *pulumirpc.DeleteRequest) (*p
 	}
 
 	// Create a new destroy diff.
-	diff := p.tf.NewDestroyDiff(ctx, string(t), shim.TimeoutOptions{
```

Comment:
This looks like a fairly big change? Isn't urn.Type() a pulumi type, while this requires a TF type?
What does this affect?

Reply:
It's a bug introduced recently in https://github.com/pulumi/pulumi-terraform-bridge/commit/39a92a469cf6ec3452fdd446d26119852909d6af specifically to unlock this PR and previously unused/untested.

Reply:
Looks like a refactor to me wrt the `t` parameter - https://github.com/pulumi/pulumi-terraform-bridge/commit/39a92a469cf6ec3452fdd446d26119852909d6af#diff-1dee36d7f4752b488fc6d27ff19bd5e68277063de183d75ef29a02b8bf989045R1332

I don't see how that commit introduces a bug there?

Reply:
The refactor wasn't tested or used. NewDestroyDiff under default options ignores `t`. The intent though was to pass the Terraform resource name in that parameter, similarly to other methods on shim.Provider. Not the Pulumi type.